src/olympia/reviewers/utils.py

```diff
@@ -1383,7 +1382,66 @@ def confirm_auto_approved(self):
 
     def reject_multiple_versions(self):
         """Reject a list of versions.
-        Note: this is used in blocklist.utils.disable_addon_for_block for both
+        This should only be used by direct reviewer actions (human_review=True).
+        See auto_reject_multiple_versions also, that can handle delayed rejections from
+        different reviewers"""
+        assert self.human_review is True
+        channel = self.version.channel if self.version else None
+        self.version = None
+        self.file = None
+
+        if self.data.get('delayed_rejection'):
+            action_id = (
+                amo.LOG.REJECT_CONTENT_DELAYED
+                if self.content_review
+                else amo.LOG.REJECT_VERSION_DELAYED
+            )
+            log.info(
+                'Marking %s versions %s for delayed rejection'
+                % (self.addon, ', '.join(str(v.pk) for v in self.data['versions']))
+            )
+        else:
+            action_id = (
+                amo.LOG.REJECT_CONTENT
+                if self.content_review
+                else amo.LOG.REJECT_VERSION
+            )
+            log.info(
+                'Making %s versions %s disabled'
+                % (self.addon, ', '.join(str(v.pk) for v in self.data['versions']))
+            )
+
+        for version in self.data['versions']:
+            # Clear needs human review flags on rejected versions, we
+            # consider that the reviewer looked at them before rejecting.
+            self.clear_specific_needs_human_review_flags(version)
+            self.set_human_review_date(version)
+
+        # A human rejection (delayed or not) implies the next version in the
+        # same channel should be manually reviewed.
+        auto_approval_disabled_until_next_approval_flag = (
+            'auto_approval_disabled_until_next_approval'
+            if channel == amo.CHANNEL_LISTED
+            else 'auto_approval_disabled_until_next_approval_unlisted'
+        )
+        AddonReviewerFlags.objects.update_or_create(
+            addon=self.addon,
+            defaults={auto_approval_disabled_until_next_approval_flag: True},
+        )
```

Comment:
Should this be part of the `ContentAction` ? It feels like a consequence of the action, something that we wouldn't do if the action is "cancelled" after being held up for approval.

Reply:
My thinking was that in either case we'd want to stop any further auto approvals. It also prevents any approvals while the rejection is held.

Reply:
mmm yeah preventing any approvals while the rejection is held certainly makes sense.